pkg/ruler/ruler_pagination.go

```diff
@@ -0,0 +1,42 @@
+package ruler
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+)
+
+type GroupStateDescs []*GroupStateDesc
```

Comment:
Should this be called PaginedGroupStates or something like that since this is only used in the pagination context?

Reply:
It will now need to be public because it is used in integ tests